### ISSUE TITLE
6.0.0-6.0.5 broken support for multiple swagger.json definitions

### ISSUE THREAD

**Issue by niker** [domaindrivendev/Swashbuckle.AspNetCore#2018]
Swashbuckle 6.0.0 - 6.0.5 has broken multiple calls to UseSwagger().

This works in 5.x.x to have both modern API and V2 API hosted side by side:

```
app.UseSwagger();
app.UseSwagger(o =>
{
    o.RouteTemplate = $"/swagger/{{documentname}}/swagger-oas2.json";
    o.SerializeAsV2 = true;   
});
```

The same code in 6.0.5 will not work, the second UseSwagger declaration will override the first and the first json will not be hosted.

Full RouteTemplate specification for the first declaration didn't work either, it is just ignored.

**Comment by alanwales** [comment]
We have experienced the same issue and was caught by our tests. We host multiple open api documents and with Swashbuckle version 6 they are no longer generated

**Comment by domaindrivendev** [comment]
This is similar to #1996, which was resolved by #2016. That MR didn't include the change for the `Swagger` middleware, so I just created 89d75c106daac5ba9dca2f4c5a577c2f53d4e219 for that. Will be available as of `6.0.7` which is running in CI right now.

**Comment by niker** [comment]
The problem is fixed in 6.0.7, thank you.

**Comment by alanwales** [comment]
Same for us, thanks a lot!
